### DISCUSSION TITLE
vike express react-helmet example

### DISCUSSION THREAD

**Discussion by rhildred** [vikejs/vike#1175] [Discussion]
In the docs there is a 1 line mention of react-helmet. There is also an example of using vite express middleware and vike. I really want to use these together to serve react pages created with teleport.hq. I tried ReactDOMServer.renderToString but it doesn't render imported css. I also tried renderToStream  from react-streaming/server. I can't figure out how to get the pipe it returns in node to go to a string and whether there is anything else that needs to be done to get the css to render with the html. Any example or other guidance would really help.

**Comment by brillout** [comment]
> We recommend to use such library only if you have a rationale: the aforementioned solutions are simpler and work for most use cases.

What's your motivation for using react-helmet?

I'm not familiar with readct-helmet, so I can't help much on that end. In principle, since Vike gives you full control over the React integration there shouldn't be a problem.

**Reply by rhildred** [reply]
Hi Rom:

Thanks for getting back to me so quickly. I am working on an integration with teleportHq. The code that they generate uses react-helmet. I found that the cloudflare worker react example was like what I want to do with the express server and in production with cloudflare workers for micro front ends. I will work on that. Unfortunately I am working on this while I have students who are testing for me. I meet with them Wednesday so might not be able to look at this again till Thursday.

Thanks again for getting back to me so quickly.

Rich

**Reply by brillout** [reply]
>  I am working on an integration with teleportHq. 

Neat, keep me updated how that goes.

> Unfortunately I am working on this while I have students who are testing for me. I meet with them Wednesday so might not be able to look at this again till Thursday.

👍 No rush.
